pdu/Src/cooling.c

```diff
@@ -0,0 +1,93 @@
+#include "cooling.h"
+
+#include <stdbool.h>
+#include "FreeRTOS.h"
+#include "task.h"
+#include "state_machine.h"
+#include "controlStateMachine.h"
+#include "FreeRTOS_CLI.h"
+#include "debug.h"
+#include "pdu_can.h"
+#include "watchdog.h"
+#include "canReceive.h"
+#include "pdu_dtc.h"
+#include "LTC4110.h"
+
+#define MAX(a, b) ((a) > (b) ? (a) : (b))
+#define SCALE_TEMP_PERCENT(temp, max_temp) (((temp) - COOLING_AMBIENT_TEMP_C) / ((max_temp) - COOLING_AMBIENT_TEMP_C))
+
+void coolingOff(void) {
+    DEBUG_PRINT("Turning cooling off\n");
+    PUMP_LEFT_DISABLE;
+    PUMP_RIGHT_DISABLE;
+    FAN_LEFT_DISABLE;
+    FAN_RIGHT_DISABLE;
+}
+
+void coolingOn(void) {
+    DEBUG_PRINT("Turning cooling on\n");
+    PUMP_LEFT_ENABLE;
+    PUMP_RIGHT_ENABLE;
+    FAN_LEFT_ENABLE;
+    FAN_RIGHT_ENABLE;
+}
+
+void coolingTask(void *pvParameters) {
+    if (registerTaskToWatch(COOLING_TASK_ID, 2*COOLING_TASK_PERIOD_MS, false, NULL) != HAL_OK)
+    {
+        ERROR_PRINT("Failed to register cooling task with watchdog!\n");
+        Error_Handler();
+    }
+
+    TickType_t xLastWakeTime = xTaskGetTickCount();
+
+    uint32_t cooling_state = COOL_STATE_WAIT;
+
+    while(1)
+    {
+        // Only cool if motor controller on and EM enabled
```

Comment:
I think if weve been to EM we should latch and keep cooling on (since we turn off the MC so cant get its temp readings) but we will get cooling temp from one of the new sensors on the DAQ so maybe add the logic to check that but leave the function blank and leave it as a TODO

Reply:
Def keep the condition about DCDC on tho. Would be tempted to change the name of that variable to clearly indicate what the variable means like dcdcOn